chain-patrol-ts/src/agent.ts

```diff
@@ -0,0 +1,99 @@
+import { BlockEvent, Finding, Initialize, HandleBlock } from "forta-agent";
+import { fetchApiInfo, getCurrentDateInYyyyMmDD } from "./utils";
+import { ApiInfo, Asset, AssetDetails, UnalertedAsset } from "./types";
+import { createBlockedAssetFinding } from "./findings";
+import AssetFetcher from "./fetcher";
+import {
+  ETHEREUM_BLOCKS_IN_ONE_DAY,
+  MAX_ASSET_ALERTS_PER_BLOCK,
+  ASSET_BLOCKED_STATUS,
+  INIT_API_QUERY_DATE,
+  ASSET_DETAILS_URL,
+  ASSET_LIST_URL,
+} from "./constants";
+
+let assetFetcher: AssetFetcher;
+let apiQueryStartDate: string;
+
+// Bots are allocated 1GB of memory, so storing
+// `UnalertedAsset`s won't be an issue. Especially
+// since entries will be cleared after alerted.
+const unalertedAssets: UnalertedAsset[] = [];
+
+async function createAssetFetcher(
+  apiKey: string,
+  assetListUrl: string,
+  assetDetailsUrl: string,
+  assetStatus: string
+): Promise<AssetFetcher> {
+  return new AssetFetcher(apiKey, assetListUrl, assetDetailsUrl, assetStatus);
+}
+
+export function provideInitialize(
+  fetchApiInfo: () => Promise<ApiInfo>,
+  assetListUrl: string,
+  assetDetailsUrl: string,
+  assetStatus: string,
+  createAssetFetcher: (
+    apiKey: string,
+    assetListUrl: string,
+    assetDetailsUrl: string,
+    assetStatus: string
+  ) => Promise<AssetFetcher>,
+  initApiQueryDate: string
+): Initialize {
+  return async () => {
+    const { API_KEY: apiKey } = await fetchApiInfo();
```

Comment:
adding a try-catch here could be useful. in case something goes wrong, a message like "error during initialization" can be more clear.

Reply:
Do you mean have the call to `fetchApiInfo()` be wrapped in a `try/catch`? If so, `fetchApiInfo` has a `try/catch` inside of it, so if there were an issue when fetching the API key _there_, it would bubble up.

Does that serve the point of what you're getting at or am I missing/overlooking something?

Reply:
yes that's right. I'm not sure what the best practice is but it is what I mean. I think we can resolve this 👍